bond/bond.go

```diff
@@ -225,6 +229,41 @@ func setLinksinNetNs(bondConf *bondingConfig, nspath string, releaseLinks bool)
 	return nil
 }
 
+func validateMTU(slaveLinks []netlink.Link, mtu int) error {
+	if mtu == 0 {
+		return nil
+	}
+	if mtu < 68 {
+		return fmt.Errorf("Invalid bond MTU value (%+v), should be 68 or bigger", mtu)
+	}
+	netHandle, err := netlink.NewHandle()
+	if err != nil {
+		return fmt.Errorf("Failed to create a new handle, error: %+v", err)
+	}
+	defer netHandle.Delete()
+
+	pfLinks, err := netHandle.LinkList()
+	if err != nil {
+		return 	fmt.Errorf("Failed to lookup physical functions links, error: %+v", err)
+	}
+	for _, pfLink := range pfLinks {
```

Comment:
how you are able to get the PF info? you are inside the pod namespace already no?

Reply:
Note that a cni does not work inside a namespace. You have to enter the namespace yourself.
One of the input parameters the CNI binary is getting is the CNI_NETS environment variable, which points us to the namespace for the container it is to connect.
We are getting the netlink handle for the pf without specifying the ns ([here](https://github.com/k8snetworkplumbingwg/bond-cni/blob/296681abbb01c9ab35c0f38af9a1769761d1e931/bond/bond.go#L239)), so we operate with no namespace here.
The slaves are retrieved in  getLinkObjectsFromConfig() which takes a handle created using a ns ([here](https://github.com/k8snetworkplumbingwg/bond-cni/blob/296681abbb01c9ab35c0f38af9a1769761d1e931/bond/bond.go#L278)).